Source/Blazorise.Bootstrap/BootstrapClassProvider.cs

```diff
@@ -310,6 +310,11 @@ public override string LabelType( LabelType labelType )
 
     public override string FieldLabel( bool horizontal ) => horizontal ? "col-form-label" : null;
 
+    public override string ValidationIndicator( bool validationIndicator )
+        => validationIndicator
+            ? "b-required"

```

Comment:
name it as "b-field-required", or "b-field-label-required", or "b-input-required"?

Reply:
Initial I thought b-required could be used generally whenever we wanted to tell something was required be it a label, input ,etc... 
But we can specify it, in that case, I like `b-field-label-required`

Reply:
Actually, I think we should try to follow each of the provider naming strategies. For example, in BS4 `form-group` represents a Field, and then it could be `form-group-required`, or `form-control-required` if we want to target inputs. You get the idea.

![image](https://github.com/Megabit/Blazorise/assets/900302/c7bcb967-ebe7-447b-bf5e-12fd1e1661c2)

Reply:
hmm You're just looking at bs, each have their own. What would be the benefit? 
Additionally we would to potentially have more code / ceremony if we want to target other components from the `FieldLabel`? 
And that gives the sense that a lot more can happen since the class is more general, while we're just adding functionality specific to the label.

Reply:
Well, I tend to use our custom `.b-` classnames as little as possible. Also, I want to stick to the naming of each provider as much as I can.

Reply:
Well we need to differentiate what's custom made, for both our sanity and the users...
Imagine in a few years time, we have a bug in the feature, and you read the class `form-group-required`, you're going to think it's a bs class right away.

Reply:
That's a future us problem 🤣

Reply:
Ok then. So just to reiterate we'll be setting `ValidationIndicator` on the `FieldLabel` and then adding the class to the parent div if it exists (which should be the Parent `Field`), `<div class="form-group form-group-required"`. This is it right?

Reply:
Considering an API that you started

```razor
<Field>
    <FieldLabel ValidationIndicator>
        Name
    </FieldLabel>
    <FieldBody>
        <TextEdit Placeholder="Name" />
    </FieldBody>
</Field>
```

I would say that it will need to be on the `<label>` instead. But then again, I don't like that API. I am not sure if I want to have it on a `FieldLabel`.

What if we add `Required` on the `Validation` component? So the API would be:

```razor
<Validation Required>
    <Field>
        <FieldLabel ValidationIndicator>
            Name
        </FieldLabel>
        <FieldBody>
            <TextEdit Placeholder="Name" />
        </FieldBody>
    </Field>
</Validation>
```

If the required asterisk is only going to be used along with the Validation, than it makes sense. Otherwise, if it can be used without validation, then it can stay on FieldLabel.

Reply:
Well in my mind It's just a visual indicator whether there's validation or not or even if the user does validation some other way. That's why it's in the label.

If you put Required in the Validation component then one could assume that it would automatically validate emptyness?
And you are saying that in this case there would be no ValidationIndicator parameter?

Reply:
> Well in my mind It's just a visual indicator whether there's validation or not or even if the user does validation some other way. That's why it's in the label.
> 
> If you put Required in the Validation component then one could assume that it would automatically validate emptyness? And you are saying that in this case there would be no ValidationIndicator parameter?

Makes sense. You're right, it should stay on FieldLabel. I would just rename it to `RequiredIndicator`.

Reply:
> * Adds the class to the parent div if it exists (which should be the Parent `Field`),
>   `<div class="form-group form-group-required">`. This is it, right?

No, don't add it to the parent. That would complicate it as you would have to know if there is a child component Field >FieldLabel

It's easier to have `RequiredIndicator` on the `FieldLabel`, and assign the class to its element.

> The naming must be in par with whatever's defined for the `Field` for the current provider:
> i.e for bs: `form-group` => `form-group-required`

See what is already on the label element. If there is `form-label` like in BS5, then you can add `form-label-required`. If there is no label class available, like on some provider then think of a name that could be best for that provider.

Reply:
> It's easier to have RequiredIndicator on the FieldLabel, and assign the class to its element.

Well my point all along that we were adding unecessary complexity... that's what's already implemented... Will change just the Parameter and class name's then.

Reply:
Done. Please review again at your earliest convenience.

Reply:
I was trying to come up with a way to automate it. But in the end, you were right all along :)

Reply:
Yeah, the FieldLabel could somehow detect if its placed inside of a `Validation` component and if the validated field is required. But not sure if it would take too long to do. Might be something for the future.